### ISSUE TITLE
Traefik2 not creating routers and services using labels, only File provider

### ISSUE THREAD

**Issue by joshoram80** [filebrowser/filebrowser#1907]
Regarding ["Unable to use traefik as a proxy for filebrowser docker container #1667"](https://github.com/filebrowser/filebrowser/issues/1667)

This still seems to be happening using the latest version of traefik and filebrowser images. Funnily enough. It works fine when setup using the File provider, but not labels

> http:
>   routers:
>     fb:
>       entryPoints:
>         - https # entrypoint defined above
>       middlewares:
>         - chain-no-auth@file
>       service: fb
>       rule: Host(`fb.<domain>`) # Not my actual domain
> 
>   services:
>     fb:
>       loadBalancer:
>         servers:
>           - url: http://fb:8088
>         passHostHeader: true

I have 20 other services and routers setup with Docker and Traefik, but Filebrowser is the only one that plays up.

**Comment by o1egl** [comment]
You need to check your traefik config or rise an issue on the traefik project.

**Comment by djeinstine** [comment]
Hi guys. I think this should be opened again. I can't be the only one who is using filebrowser behind traefik. filebrowser is the only container that doesn't work right now. Anyhow, I think it was not a good decision to use port 80. At least change the port to something generic like 8080. Saying "its not us, its you" is bad form and a non-answer.

**Comment by akarub** [comment]
> Hi guys. I think this should be opened again. I can't be the only one who is using filebrowser behind traefik. filebrowser is the only container that doesn't work right now. Anyhow, I think it was not a good decision to use port 80. At least change the port to something generic like 8080. Saying "its not us, its you" is bad form and a non-answer.

I'm having the same problem. This [solution ](https://github.com/filebrowser/filebrowser/issues/1667#issuecomment-969248664)didn't work for me. I simply gave up trying to run filebrowser behind traefik.

**Comment by jlopez994** [comment]
I got it working by adding this configuration to the .filebrowser.json

```
 {
   "address": "0.0.0.0",
   "port": 80
 }
```

And these are my labels in docker-compose.yml

```
labels:
  - "traefik.enable=true"
  - "traefik.http.routers.filebrowser.rule=Host(`host...`)"
  - "traefik.http.routers.filebrowser.entrypoints=websecure"
  - "traefik.http.routers.filebrowser.tls.certresolver=cloudflare"
```

**Comment by drewstopherlee** [comment]
I am also having problems related to Traefik, and the issue is most certainly NOT Traefik. Two out of three deployments (with identical bind mounts and identical settings.json files) are failing their healthchecks, and any container considered "unhealthy" is automatically skipped by Traefik unless you force it by using the file provider. This is definitely an issue with the healthcheck built into the container, not Traefik.

**Comment by theriaultjulien** [comment]
I have the same problem with my deployment of filebrowser with Traefik. 
I did as mentioned here: [Issue 1667](https://github.com/filebrowser/filebrowser/issues/1667#issuecomment-969248664)
Configuring the healthcheck manually in my compose seems to have fixed it.
```
healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:80/health"]
      interval: 30s
      timeout: 20s
      retries: 3
```

Now I understand if using custom ports that the healthcheck might not work, but I'm using the default port of 80. The healthcheck should be working by itself.

**Comment by drewstopherlee** [comment]
>I did as mentioned here: [Issue 1667](https://github.com/filebrowser/filebrowser/issues/1667#issuecomment-969248664) Configuring the healthcheck manually in my compose seems to have fixed it.

I tried that config and wound up having to fully disable the healthchecks for it to be picked up by Traefik, no matter what I did it always failed. Per [this comment](https://github.com/filebrowser/filebrowser/issues/3085#issuecomment-2031357247), it seems to have been fixed with #3087 and we should "wait for the next release." The joys of FOSS lol.

**Comment by sargonas** [comment]
Oh thank goodness this isn't just me! My labels stopped working, but ONLY for filebrowser, and I was pulling out my hair trying to hase it down on the Traefik side. So relieved to see it's related to FB instead, at least I know I'm not going insane@
